src/main/java/Dude.java

```diff
@@ -0,0 +1,204 @@
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class  Dude {
+
+    static TaskList taskList = new TaskList();
+
+    static final String[] supported_commands = {"bye", "list", "mark", "unmark", "todo", "event", "deadline", "delete"};
+
+    public static void main(String[] args) {
+
+//        String logo =   "888888ba                 dP          \n" +
+//                        "88    `8b                88          \n" +
+//                        "88     88 dP    dP .d888b88 .d8888b. \n" +
+//                        "88     88 88    88 88'  `88 88ooood8 \n" +
+//                        "88    .8P 88.  .88 88.  .88 88.  ... \n" +
+//                        "8888888P  `88888P' `88888P8 `88888P'";
+//
+//        System.out.println("--------------------------------------\n");
+//        System.out.println(logo + "\n");
+//        System.out.println("--------------------------------------");
+//        System.out.println("Dude v1.0 by Tahsin Hasem.\n");
+
+        System.out.println(greet());
+
+        // Initialise Scanner to reach command line input
+        Scanner sc = new Scanner(System.in);
+        while(true){
+
+            String msg = "";
+            try {
+                msg = sc.nextLine();
+            }catch(NoSuchElementException e){
+                //this will not be handled.
+                break;
+            }
+
+            String command = "";
+            try {
+                command = get_command(msg);
+            }catch (InvalidCommandException e){
+                System.out.println(echo(e.getMessage()));
+                continue;
+            }
+
+            switch (command){
+                case "bye":
+                    System.out.println(bye());
+                    return;
+                case "list":
+                    System.out.println(list());
+                    break;
+                case "mark":
+                    System.out.println(mark_as_done(msg));
+                    break;
+                case "unmark":
+                    System.out.println(mark_as_undone(msg));
+                    break;
+                case "todo":
+                    System.out.println(handle_todo_command(msg));
+                    break;
+                case "event":
+                    System.out.println(handle_event_command(msg));
+                    break;
+                case "deadline":
+                    System.out.println(handle_deadline_command(msg));
+                    break;
+                case "delete":
+                    System.out.println(handle_delete_command(msg));
+                    break;
+            }
```

Comment:
LGTM. I like the way how you are handling your command. The code looks short and easy to understand.